app/logic/userManagement.py

```diff
@@ -33,3 +33,27 @@ def changeCurrentTerm(term):
     newCurrentTerm.save()
 
     session["current_term"] = model_to_dict(newCurrentTerm)
+
+def addTerm():
```

Comment:
There's no test for this! This is a very clear case where TDD makes sense. I suspect it would have saved you a lot of time.

Reply:
I added test for the AddNewTerm function